integration-tests/src/integrationTest/java/pl/allegro/tech/hermes/integrationtests/MetricsTest.java

```diff
@@ -257,6 +257,69 @@ public void shouldReportMetricForFilteredSubscription() {
+        final Subscription subscription = hermes.initHelper().createSubscription(
+                subscription(topic, "subscription")
+                        .withEndpoint(subscriber.getEndpoint())
+                        .withSubscriptionPolicy(
+                                subscriptionPolicy()
+                                        .withMessageTtl(3600)
+                                        .withInflightSize(1)
+                                        .build()
+                        )
+                        .withFilter(filterMatchingHeaderByPattern("Trace-Id", "^vte.*"))
+                        .build()
+        );
+        TestMessage unfiltered = TestMessage.of("msg", "unfiltered");
+        TestMessage filtered = TestMessage.of("msg", "filtered");
+
+        // when
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), filtered.body(), header("Trace-Id", "otherTraceId"));
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), filtered.body(), header("Trace-Id", "otherTraceId"));
+
+        // then
+        waitAtMost(Duration.ofSeconds(10)).untilAsserted(() ->
+                hermes.api().getConsumersMetrics()
+                        .expectStatus()
+                        .isOk()
+                        .expectBody(String.class)
+                        .value((body) -> assertThatMetrics(body)
+                                .contains("hermes_consumers_subscription_inflight")
+                                .withLabels(
+                                        "group", topic.getName().getGroupName(),
+                                        "subscription", subscription.getName(),
+                                        "topic", topic.getName().getName()
+                                )
+                                .withValue(0.0)
+                        )
+        );
+
+        // when
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), unfiltered.body(), header("Trace-Id", "vte12"));
+        hermes.api().publishUntilSuccess(topic.getQualifiedName(), unfiltered.body(), header("Trace-Id", "vte12"));
+
+        // then
+        waitAtMost(Duration.ofSeconds(10)).untilAsserted(() ->
+                hermes.api().getConsumersMetrics()
+                        .expectStatus()
+                        .isOk()
+                        .expectBody(String.class)
+                        .value((body) -> assertThatMetrics(body)
+                                .contains("hermes_consumers_subscription_inflight")
+                                .withLabels(
+                                        "group", topic.getName().getGroupName(),
+                                        "subscription", subscription.getName(),
+                                        "topic", topic.getName().getName()
+                                )
+                                .withValue(1.0)
+                        )
```

Comment:
why 1.0 instead of 2.0?

Reply:
Because previously we set `subscriptionPolicy().withInflightSize(1)` and we want to be sure, that this value wouldn't be exceeded when there are more messages and there were messages filtered previously.